test/EtomicSwap.js

```diff
@@ -274,6 +427,83 @@ contract('EtomicSwap', function(accounts) {
         await this.swap.receiverSpend(id, web3.utils.toWei('1'), secretHex, this.token.address, accounts[0], { from: accounts[1], gasPrice }).should.be.rejectedWith(EVMThrow);
     });
 
+    it('should allow receiver to spend ERC721 payment by revealing a secret', async function () {
+        const lockTime = await currentEvmTime() + 1000;
+        const tokenId = 1; // Assuming token ID 1 is minted to accounts[0]
+
+        const data = web3.eth.abi.encodeParameters(
+            ['bytes32', 'address', 'address', 'bytes20', 'uint64'],
+            [id, accounts[1], this.erc721token.address, secretHash, lockTime]
+        );
+        // Call safeTransferFrom directly to transfer the token to the EtomicSwap contract
+        await this.erc721token.safeTransferFrom(accounts[0], this.swap.address, tokenId, data).should.be.fulfilled;
+
+        // Check the ownership of the token before receiver spend payment - should be owned by swap contract
+        const tokenOwnerBeforeReceiverSpend = await this.erc721token.ownerOf(tokenId);
+        assert.equal(tokenOwnerBeforeReceiverSpend, this.swap.address);
+
+        // Attempt to spend with invalid secret - should fail
+        await this.swap.receiverSpendErc721(id, zeroAddr, this.erc721token.address, tokenId, accounts[0], { from: accounts[1] }).should.be.rejectedWith(EVMThrow);
```

Comment:
`zeroAddr` is 20 bytes instead of 32, please use 32 bytes argument for 100% correctness :slightly_smiling_face:

Reply:
added invalidSecretHex